client/internal/engine.go

```diff
@@ -244,7 +244,7 @@ func (e *Engine) Stop() error {
 
 	err := e.removeAllPeers()
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to remove all peers: %s", err)
```

Comment:
```suggestion
		return fmt.Errorf("remove all peers: %w", err)
```

Let's wrap the error, also "failed to" doesn't add any information, it's already an error.